In [14]:
import numpy as np

#Error Multiplied By A Number
#Q = cA
def rule1(c, dA):
    dQ = np.abs(c)*dA
    return dQ

def rule1(c, dA, dB):
    print("hi")

#Error with a Power (includings multiplication of a constant)
#Q = cA^m
def rule2(c, A, dA, m):
    dQ = np.abs(c*m*A**(m-1))*dA
    return dQ

#Addition/Subtraction
#Q = A + B or Q = A - B
def rule3(dA, dB):
    dQ = np.sqrt(dA**2 + dB**2)
    return dQ

#Multiplication between Errors (including multiplication of a constant and Powers)
#Q = c*A^m*B^n
def rule4(Q, A, dA, m, B, dB, n):
    dQ = np.abs(Q)*np.sqrt((m*(dA/A))**2+(n*(dB/B))**2)
    return dQ

def rule4_3vars(Q, A, dA, m, B, dB, n, C, dC, o):
    dQ = np.abs(Q)*np.sqrt((m*(dA/A))**2+(n*(dB/B))**2+o*(dC/C))
    return dQ

Fc = rule4_3vars(.838, .177, .01, 1, .2083, .1, 1, 1.3011, 0.008225, -2)
print(Fc)

0.3939692176149074


Fc above is the program running to solve $\Sigma F_{c}$ from the previous lab. I'm getting a result on here 1 magnitude higher than I did last week, but I'm fairly certain that's due to me messing up last week.

In [22]:
x = [1.1, 1.3, 1.4, 0.9, 0.95, 1.05]
x_avg = np.average(x)

print(x_avg)

dx = np.std(x)/np.sqrt(6)

print(dx)

1.1166666666666667
0.073282810879294


This means that the error in the mean of the list is $1.1167\pm0.0733$

Rule 3:
$\delta Q = \sqrt{\delta A^2 + \delta B^2}$


Rotation Case from last Lab:
$\Sigma F_{c} = \frac{4\pi^2rm}{T^2}$

$\Sigma F_{c} = 4\pi^2rmT^{-2}$